spec/features/search_spec.rb

```diff
@@ -30,4 +30,82 @@
     expect(page).to have_content(email_match.email)
     expect(page).not_to have_content(mismatch.email)
   end
+
+  scenario "admin searches using a model scope", :js do
+    query = "scope:subscribed"
+    subscribed_customer = create(
+      :customer,
+      name: "Dan Croak",
+      email_subscriber: true)
+    other_customer = create(
+      :customer,
+      name: "Foo Bar",
+      email_subscriber: false)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+
+    expect(page).to have_content(subscribed_customer.name)
+    expect(page).not_to have_content(other_customer.name)
+  end
+
+  scenario "ignores malicious scope searches", :js do
+    query = "scope:destroy_all"
+    customer = create(
+      :customer,
+      name: "FooBar destroy_all: user",
+      email_subscriber: false)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+    expect(page).to have_content(customer.name)
+  end
+
+  scenario "admin searches into a model scope", :js do
+    searching_for = "Lua"
+    query = "scope:subscribed #{searching_for}"
+    subscribed_frog = create(
+      :customer,
+      name: "Kermit Croaks",
+      email_subscriber: true)
+    subscribed_cat = create(
+      :customer,
+      name: "#{searching_for} Miaus",
+      email_subscriber: true)
+    unsubscribed_cat = create(
+      :customer,
+      name: "#{searching_for} Doe",
+      email_subscriber: false)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(subscribed_cat.name)
+    end
+    expect(page).not_to have_content(subscribed_frog.name)
+    expect(page).not_to have_content(unsubscribed_cat.name)
+  end
+
+  scenario "admin clicks a scope button", :js do
+    subscribed = create(
+      :customer,
+      name: "Lua Miaus",
+      email_subscriber: true)
+    unsubscribed = create(
+      :customer,
+      name: "John Doe",
+      email_subscriber: false)
+
+    visit admin_customers_path
+    click_on 'subscribed'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.